specification/billing/resource-manager/Microsoft.Billing/stable/2024-04-01/agreement.json

```diff
@@ -0,0 +1,321 @@
+            "ESignOffline",
+            "Implicit",
+            "Offline",
+            "PhysicalSign"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "AcceptanceMode",
+            "modelAsString": true
+          }
+        },
+        "agreementLink": {
+          "description": "The URL to download the agreement.",
+          "type": "string",
+          "readOnly": true
+        },
+        "billingProfileInfo": {
+          "description": "The list of billing profiles associated with agreement and present only for specific agreements.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BillingProfileInfo"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": []
+        },
+        "category": {
+          "description": "The category of the agreement.",
+          "enum": [
+            "Other",
+            "AffiliatePurchaseTerms",
+            "IndirectForGovernmentAgreement",
+            "MicrosoftCustomerAgreement",
+            "MicrosoftPartnerAgreement",
+            "UKCloudComputeFramework"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "Category",
+            "modelAsString": true
+          }
+        },
+        "displayName": {
+          "description": "The name of the agreement signed by a customer.",
+          "type": "string",
+          "readOnly": true
+        },
+        "effectiveDate": {
+          "format": "date-time",
+          "description": "The date from which the agreement is effective.",
+          "type": "string",
+          "readOnly": true
+        },
+        "expirationDate": {
+          "format": "date-time",
+          "description": "The date when the agreement expires.",
+          "type": "string",
+          "readOnly": true
+        },
+        "participants": {
+          "description": "The list of participants that participates in acceptance of an agreement.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Participant"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": []
+        },
+        "status": {
+          "description": "The current status of the agreement.",
+          "type": "string",
+          "readOnly": true
+        },
+        "leadBillingAccountName": {
+          "description": "The ID of the lead billing account if this agreement is part of the Customer Affiliate Purchase Terms.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "BillingProfileInfo": {
+      "description": "Details about billing profile associated with agreement and available only for specific agreements.",
+      "type": "object",
+      "properties": {
+        "billingAccountId": {
+          "description": "The fully qualified ID that uniquely identifies a billing account.",
```

Comment:
> fully qualified ID [](http://example.com/codeflow?start=29&length=19)

just wondering what these look like - are these guids or something? resource ids?

Reply:
examples did sort of make it clear.

Reply:
Depends on the channel. Some are numbers (123456789), some are guids, and others are a compound of a GUID:GUID_date

Reply:
okay. So what does fully qualified even mean? Just wondering.